examples/hybrid_jobs/6_QNSPSA_optimizer_with_embedded_simulator/qnspsa_with_embedded_simulator.ipynb

```diff
@@ -0,0 +1,1360 @@
+   "source": [
+    "Script `source_scripts/benchmark_qaoa_converge_speed` tests the performance of the four optimizers with the QAOA maxcut problem described above. As Pennylane does not provide a directly implementation on SPSA, here my SPSA optimzer is created by disabling the Fisher tensor metric estimation in QN-SPSA. In the script, I have adjusted the learning rates to optimize each of the optimizers' performance (tested separately). Considering the stochastic nature of SPSA and QN-SPSA, they are each repeated `spsa_repeats` times to capture the average performance. The below cell confirms that the Job works."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 8,
+   "id": "6e8bacaa",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "WARNING:braket.jobs.local.local_job_container_setup:Using the short-lived AWS credentials found in session. They might expire while running.\n"
+     ]
+    },
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Boto3 Version:  1.22.7\n",
+      "Beginning Setup\n",
+      "Running Code As Subprocess\n",
+      "0.23.0\n",
+      "{'n_qubits': '4', 'edges': '5', 'depth': '2', 'shots': '1000', 'seed': '1', 'load_init_config': 'False', 'max_iter': '100', 'spsa_repeats': '2'}\n",
+      "Using local simulator:  Lightning Qubit PennyLane plugin\n",
+      "\n",
+      "Gradient descent optimizer:\n",
+      "Step 0:  loss = -2.55\n",
+      "Step 20:  loss = -3.17\n",
+      "Step 40:  loss = -3.23\n",
+      "Step 60:  loss = -3.21\n",
+      "Step 80:  loss = -3.21\n",
+      "\n",
+      "Quantum natural gradient optimizer:\n",
+      "Step 0:  loss = -2.55\n",
+      "Step 20:  loss = -3.20\n",
+      "Step 40:  loss = -3.25\n",
+      "Step 60:  loss = -3.24\n",
+      "Step 80:  loss = -3.20\n",
+      "\n",
+      "QN-SPSA optimizer:\n",
+      "Trace 0:\n",
+      "Step 0:  loss = -2.54\n",
+      "Step 20:  loss = -2.72\n",
+      "Step 40:  loss = -3.06\n",
+      "Step 60:  loss = -3.11\n",
+      "Step 80:  loss = -3.15\n",
+      "Trace 1:\n",
+      "Step 0:  loss = -2.56\n",
+      "Step 20:  loss = -2.91\n",
+      "Step 40:  loss = -3.06\n",
+      "Step 60:  loss = -3.12\n",
+      "Step 80:  loss = -3.54\n",
+      "\n",
+      "SPSA optimizer:\n",
+      "Trace 0:\n",
+      "Step 0:  loss = -2.62\n",
+      "Step 20:  loss = -3.14\n",
+      "Step 40:  loss = -3.06\n",
+      "Step 60:  loss = -3.16\n",
+      "Step 80:  loss = -3.07\n",
+      "Trace 1:\n",
+      "Step 0:  loss = -2.55\n",
+      "Step 20:  loss = -3.14\n",
+      "Step 40:  loss = -2.99\n",
+      "Step 60:  loss = -3.18\n",
+      "Step 80:  loss = -3.19\n",
+      "Training Successful!!\n",
+      "Code Run Finished\n",
+      "CPU times: user 315 ms, sys: 111 ms, total: 427 ms\n",
+      "Wall time: 1min 31s\n"
+     ]
+    }
+   ],
+   "source": [
+    "%%time\n",
+    "n_qubits = 4\n",
```

Comment:
Why specify hyperparameters here instead of directly below when constructing the dict?

Reply:
This was initially for stringifying the parameters, and it is no longer needed. Will change to what you suggested.

Reply:
I often want to re-use the variables in defining the Jobs name. See cell 16 as an example. I also recall this was how sagemaker examples were initially presented. But I don't insist on this.